storey/flow.py

```diff
@@ -31,32 +33,35 @@ def run(self):
         for outlet in self._outlets:
             outlet.run()
 
-    async def _do(self, element):
+    async def _do(self, event):
         raise NotImplementedError
 
-    async def _do_downstream(self, element):
-        if element is _termination_obj:
+    async def _do_downstream(self, event):
+        if event is _termination_obj:
             termination_result = await self._outlets[0]._do(_termination_obj)
             for i in range(1, len(self._outlets)):
                 termination_result = self._termination_result_fn(termination_result, await self._outlets[i]._do(_termination_obj))
             return termination_result
         tasks = []
         for i in range(len(self._outlets)):
-            tasks.append(asyncio.get_running_loop().create_task(self._outlets[i]._do(element)))
+            tasks.append(asyncio.get_running_loop().create_task(self._outlets[i]._do(event)))
         for task in tasks:
             await task
 
 
+Event = collections.namedtuple('Event', 'element key time')
+
+
 class FlowController:
     def __init__(self, emit_fn, await_termination_fn):
         self._emit_fn = emit_fn
         self._await_termination_fn = await_termination_fn
 
-    def emit(self, element):
-        self._emit_fn(element)
+    def emit(self, element, key=None, event_time=time.time()):
```

Comment:
the default value of time only get's called once, change it to `=None` and set it to now in the function